### ISSUE TITLE
Supervisor not updating to supervisor-2021.04.0

### ISSUE THREAD

**Issue by FrankElias** [home-assistant/supervisor#2812]
### Describe the issue you are experiencing

Hassio reports "Your installation is unhealthy" ==> supervisor update fails with

```
21-04-14 22:03:46 INFO (MainThread) [supervisor.supervisor] Fetching AppArmor profile https://version.home-assistant.io/apparmor.txt
21-04-14 22:03:46 INFO (MainThread) [supervisor.host.apparmor] Adding/updating AppArmor profile: hassio-supervisor
21-04-14 22:03:46 INFO (MainThread) [supervisor.host.services] Reloading local service hassos-apparmor.service
21-04-14 22:03:46 INFO (MainThread) [supervisor.supervisor] Update Supervisor to version 2021.04.0
21-04-14 22:03:46 INFO (SyncWorker_5) [supervisor.docker.interface] Downloading docker image homeassistant/aarch64-hassio-supervisor with tag 2021.04.0.
21-04-14 22:04:01 ERROR (SyncWorker_5) [supervisor.docker.interface] Can't install homeassistant/aarch64-hassio-supervisor:2021.04.0 -> 500 Server Error for http+docker://localhost/v1.40/images/create?tag=2021.04.0&fromImage=homeassistant%2Faarch64-hassio-supervisor: Internal Server Error ("Get "https://registry-1.docker.io/v2/": net/http: request canceled while waiting for connection (Client.Timeout exceeded while awaiting headers)").
21-04-14 22:04:01 ERROR (MainThread) [supervisor.supervisor] Update of Supervisor failed!
21-04-14 22:04:19 WARNING (MainThread) [supervisor.host.network] Can't update connectivity information: Error: Timeout was reached
```

### What is the used version of the Supervisor?

supervisor-2021.03.9

### What type of installation are you running?

Home Assistant OS
Linux version 5.4.83-v8 (builder@3b343e44eee4) (gcc version 9.3.0 (Buildroot -gdd4918c8)) #1 SMP PREEMPT Thu Apr 1 18:30:43 UTC 2021
Machine model: Raspberry Pi 4 Model B Rev 1.4

### Which operating system are you running on?

Home Assistant Operating System

### What is the version of your installed operating system?

Home Assistant OS 5.13

### What version of Home Assistant Core is installed?

core-2021.4.2

### Steps to reproduce the issue

1. Supervisor panel > System
2. Supervisor / Newest Version --> Update
3. KO with error message above

Doing a repair on su and re-launching the update results in the same problem.

![image](https://user-images.githubusercontent.com/14946431/114790114-7ccd3800-9d84-11eb-863a-a00cf45f4090.png)

...

**Comment by pvizeli** [comment]
Sounds like a network problem. There are Timeouts and routing issues. Maybe an firewall issue?

**Comment by ludeeus** [comment]
That message you usually get when your designated DNS server is too slow to respond to DNS queries.
In the Supervisor UI, set a static DNS server to 1.1.1.1(cloudflare) or 8.8.8.8 (google), after you have done that do a full reboot of the device and it should start working

**Comment by AlessioTaruffi** [comment]
I have a similar problem 
I'm on a Raspberri pi4 too 

21-04-15 18:22:34 ERROR (SyncWorker_3) [supervisor.docker.interface] Can't install homeassistant/aarch64-hassio-supervisor:2021.04.0 -> 404 Client Error for http+docker://localhost/v1.40/images/homeassistant/aarch64-hassio-supervisor:2021.04.0/json: Not Found ("no such image: homeassistant/aarch64-hassio-supervisor:2021.04.0: No such image: homeassistant/aarch64-hassio-supervisor:2021.04.0").
21-04-15 18:22:34 ERROR (MainThread) [supervisor.supervisor] Update of Supervisor failed!

Tried changind DNS both with 8888 and 1111, still same error
First time i install Homeassistant

**Comment by ludeeus** [comment]
That is not the same, your messages indicates that you have run out of space

**Comment by FrankElias** [comment]
@ludeeus : Indeed I am running **AdGuard Home** plugin. Previousely to this error I successfully upgraded the Hassio Operating System without any complains.

**Topic is resolved as following:**
 turning off AdGuard Home (the official package supported) and replacing the DNS in my router with 1.1.1.1 and 8.8.8.8 

After some waiting this did the trick and manual update was possible.
==> It seems that some part of the supervisor updates are blacklisted by AdGuard - or you have some figgly time sensitive routines in there (as mentioned Hassio OS update worked fine).
==> Maybe you want to mention in some sort of troubleshooting guide that AdGuard Home (a superb tool when you have little kids...) could interfere with an auto-update.

Also, it seems I have just found an easy way to stop auto-updates of supervisor: just block the relevant github link in AdGuard :-).  
Having already run through several re-installs from scratch due to supervisor hickups in the past years, I consider auto-update of supervisor a weak spot of HA.

Cheers and thanks for the hints,
Fronk
